src/libexpr/lexer.l

```diff
@@ -160,8 +160,10 @@ or          { return OR_KW; }
                  }
 <IND_STRING>\$\{ { PUSH_STATE(INITIAL); return DOLLAR_CURLY; }
 <IND_STRING>\'\' { POP_STATE(); return IND_STRING_CLOSE; }
-<IND_STRING>\'   {
-                   yylval->e = new ExprIndStr("'");
+<IND_STRING>(\'|\$) {
+                   char str[2] = "'";
+                   str[0] = yytext[0];
+                   yylval->e = new ExprIndStr(str);
                    return IND_STR;
                  }
 <IND_STRING>.    return yytext[0]; /* just in case: shouldn't be reached */
```

Comment:
funilly, there's now a warning that this rule is never matched.
This is not yet true for the STRING case, I suppose because there this rule (that shouldn't be matched) is matched when a string ends with \ or $ at the end of a file.